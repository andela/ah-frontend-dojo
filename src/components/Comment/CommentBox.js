// react
import React from 'react';
import PropTypes from 'prop-types';
import DeleteComment from 'components/Comment/DeleteComment';
import LikeDislikeCommentd from '../LikeDislikeComment';


const date = timestamp => (new Date(timestamp).toString().slice(0, 25));
const CommentBox = (
  {
    id, author: { username, image },
    created_at, body, articleSlug, deleteComment, authenticatedUsername, likeStatus, dislikeStatus
  }
) => (
  <div className="comment">
    <img className="comment__img" alt="author" src={image} />
    <div className="comment__header">
      <p className="comment__header__author">{username}</p>
      <span className="comment__header__timestamp">{date(created_at)}</span>
    </div>
    <div className="comment__body">{body}</div>
    <LikeDislikeCommentd
      likeStatus={likeStatus}
      dislikeStatus={dislikeStatus}
      articleSlug={articleSlug}
      commentId={id}
    />
    <div className="comment_footer">
      {' '}
      {}
    </div>
    <div className="comment__footer">
      <DeleteComment
        author={username}
        id={id}
        articleSlug={articleSlug}
        deleteComment={deleteComment}
        authenticatedUsername={authenticatedUsername}
      />
    </div>

  </div>
);

CommentBox.propTypes = {
  authenticatedUsername: PropTypes.string.isRequired,
  deleteComment: PropTypes.func.isRequired,
  articleSlug: PropTypes.string.isRequired,
  id: PropTypes.number.isRequired,
  body: PropTypes.string.isRequired,
  created_at: PropTypes.string.isRequired,
  author: PropTypes.shape({
    username: PropTypes.string.isRequired,
  }).isRequired,
  likeStatus: PropTypes.bool,
  dislikeStatus: PropTypes.bool,
};

CommentBox.defaultProps = {
  likeStatus: false,
  dislikeStatus: false,
};

export default CommentBox;
