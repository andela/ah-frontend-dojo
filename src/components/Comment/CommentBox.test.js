import React from 'react';
import { shallow } from 'enzyme';
import CommentBox from 'components/Comment/CommentBox';


describe('Comment Box', () => {
  let wrapper;
  const props = {
    author: { username: 'arthur' },
    commentId: 1,
    created_at: '2019-06-15T06:02:00.086733Z',
    body: 'my first comment',
    articleSlug: 'my_article_slug',
    deleteComment: jest.fn(),
    authenticatedUsername: 'arthur',
  };

  it('should render without crushing', () => {
    wrapper = shallow(<CommentBox
      {...props}
    />);

    expect(wrapper).toMatchSnapshot();
  });
});
