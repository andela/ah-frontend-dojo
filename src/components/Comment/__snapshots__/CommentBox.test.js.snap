// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment Box should render without crushing 1`] = `
<div
  className="comment"
>
  <img
    alt="author"
    className="comment__img"
  />
  <div
    className="comment__header"
  >
    <p
      className="comment__header__author"
    >
      arthur
    </p>
    <span
      className="comment__header__timestamp"
    >
      Sat Jun 15 2019 09:02:00 
    </span>
  </div>
  <div
    className="comment__body"
  >
    my first comment
  </div>
  <Connect(LikeDislikeComment)
    articleSlug="my_article_slug"
    commentId={1}
    dislikeStatus={false}
    likeStatus={false}
  />
  <div
    className="comment_footer"
  >
     
  </div>
  <div
    className="comment__footer"
  >
    <DeleteComment
      articleSlug="my_article_slug"
      authenticatedUsername="arthur"
      author="arthur"
      deleteComment={[MockFunction]}
      id={1}
    />
  </div>
</div>
`;
